pkg/config/generate.go

```diff
@@ -477,3 +477,34 @@ func (g *simpleGenerator) Generate(cs *api.OpenShiftManagedCluster, template *pl
 
 	return
 }
+
+// InvalidateSecrets removes all non-ca certificates, private keys and secrets from an
+// OpenShiftManagedCluster's Config
+func (g *simpleGenerator) InvalidateSecrets(cs *api.OpenShiftManagedCluster) (err error) {
```

Comment:
I don't like that the top half of this function is whitelisting things to keep, and the bottom half is blacklisting things to reset.  I think that explicitly blacklisting things to reset is a better approach throughout.  WDYT?

There's also a separate question of how we keep this function updated when types change.  That's actually a very similar problem to how we keep the admin API updated when types change.  Can you open a topic in the architecture call document so we can discuss as a team?

Reply:
+1 on just blacklisting what needs to be updated